### DISCUSSION TITLE
Would it be possible to use LMStudio as a backend (instead of Ollama)?

### DISCUSSION THREAD

**Discussion by AlloriMD** [ericmjl/llamabot#134] [Ideas]
Thanks for creating llamabot. I'm enjoying experimenting with LLMs programmatically.

Question for you -- As I understand it, Ollama is used to run the LLM model locally (like on localhost), and llamabot interacts with Ollama (presumably through something like a REST API).

Would it be possible to extend llamabot to use [LMStudio's server mode](https://lmstudio.ai/docs/basics/server)? LMStudio's UI is very clean and easy to use, arguably easier than Ollama's minimal menubar interface.

Just a thought on how to make the "onboarding process" for llamabot even easier.

**Comment by ericmjl** [comment]
@AlloriMD thanks for the question! Apologies for the delay. I think LMStudio is supported by way of LiteLLM -- you'd just need to set the api_base keyword. This should definitely go into the documentation!

**Comment by edesz** [comment]
This is now available in the docs. See section 3 from the **Get access to LLMs** in the docs ([here](https://ericmjl.github.io/llamabot/#option-3-using-local-models-with-lmstudio)). Great! :+1: 

I think this can be marked as *Answered*.
